src/hypofuzz/dashboard/test.py

```diff
@@ -47,14 +47,15 @@ def __post_init__(self) -> None:
         reports_by_worker = self.reports_by_worker
         self.reports_by_worker = {}
 
-        # TODO: use k-way merge for nlog(k)) performance, since reports_by_worker
+        # use k-way merge for nlog(k) performance, since reports_by_worker
         # is already sorted.
-        # This sorting won't matter for correctness (once we correctly insert
-        # out-of-order reports), but it will for performance, by minimizing the
-        # number of .bisect calls in the worker reports list.
-        for report in sorted(
-            itertools.chain.from_iterable(reports_by_worker.values()),
-            key=lambda r: r.elapsed_time,
+        #
+        # This sorting doesn't matter for correctness (once we correctly insert
+        # out-of-order reports), but it does for performance, by minimizing the
+        # number of .bisect calls in the worker reports list. (I have not profiled
+        # this).
```

Comment:
I'm not following here; how can we bisect if it's not sorted?

Reply:
This should be "by minimizing the work each .bisect call does"; we'll still have to bisect, but if the bisect just appends to the end (as it will if we add reports in sorted order) then that's constant vs logn.

Though to your other comment, that's not the current performance characteristic of `bisect_right`; we should fix that.